bin/aws_eks.sh

```diff
@@ -0,0 +1,170 @@
+                    eval spot_allocation_strategy='$'cluster_cloud_provisioner_node_group_${i}_spot_allocation_strategy; if [ ! -z $spot_allocation_strategy ]; then echo spot_allocation_strategy =  \""$spot_allocation_strategy"\"  >> worker_groups.tfvars; fi
+                    eval asg_desired_capacity='$'cluster_cloud_provisioner_node_group_${i}_asg_desired_capacity; if [ ! -z $asg_desired_capacity ]; then echo asg_desired_capacity =  "$asg_desired_capacity"  >> worker_groups.tfvars; fi
+                    eval asg_max_size='$'cluster_cloud_provisioner_node_group_${i}_asg_max_size; if [ ! -z $asg_max_size ]; then echo asg_max_size =  "$asg_max_size"  >> worker_groups.tfvars; fi
+                    eval asg_min_size='$'cluster_cloud_provisioner_node_group_${i}_asg_min_size; if [ ! -z $asg_min_size ]; then echo asg_min_size =  "$asg_min_size"  >> worker_groups.tfvars; fi
+                    eval root_volume_size='$'cluster_cloud_provisioner_node_group_${i}_root_volume_size; if [ ! -z $root_volume_size ]; then echo root_volume_size =  "$root_volume_size"  >> worker_groups.tfvars; fi
+                    eval kubelet_extra_args='$'cluster_cloud_provisioner_node_group_${i}_kubelet_extra_args; if [ ! -z "$kubelet_extra_args" ]; then echo kubelet_extra_args =  \""$kubelet_extra_args"\"  >> worker_groups.tfvars; fi
+                    eval spot_instance_pools='$'cluster_cloud_provisioner_node_group_${i}_spot_instance_pools; if [ ! -z $spot_instance_pools ]; then echo spot_instance_pools =  "$spot_instance_pools"  >> worker_groups.tfvars; fi
+                    eval spot_max_price='$'cluster_cloud_provisioner_node_group_${i}_spot_max_price; if [ ! -z $spot_max_price ]; then echo spot_max_price =  \""$spot_max_price"\"  >> worker_groups.tfvars; fi
+                    eval on_demand_base_capacity='$'cluster_cloud_provisioner_node_group_${i}_on_demand_base_capacity; if [ ! -z $on_demand_base_capacity ]; then echo on_demand_base_capacity =  "$on_demand_base_capacity"  >> worker_groups.tfvars; fi
+                    eval on_demand_percentage_above_base_capacity='$'cluster_cloud_provisioner_node_group_${i}_on_demand_percentage_above_base_capacity; if [ ! -z $on_demand_percentage_above_base_capacity ]; then echo on_demand_percentage_above_base_capacity =  "$on_demand_percentage_above_base_capacity"  >> worker_groups.tfvars; fi
+                    if [ "$vpc_id" == "default" ]; then
+                        echo public_ip =  \""true"\"  >> worker_groups.tfvars
+                    fi
+                    # TODO: Add bastion access `key_name`
+                    # eval name='$'cluster_cloud_provisioner_node_group_${i}_key_name; if [ ! -z $key_name ]; then echo key_name =  \""ec2-key"\"  >> worker_groups.tfvars; fi
+                    echo "}," >>  worker_groups.tfvars
+                done
+                    echo "]" >> worker_groups.tfvars
```

Comment:
1. Wrong indentation
2. Not readable. 
2.1. Copy-paste code.

Create auxiliary function and use it
For ex.
```bash
function add_tfvars {
    local tf_var_name=$1
    local yaml_var=$2
    local filename=${3:-"worker_groups.tfvars"}

    if [[! -z $yaml_var ]]; then
        echo $tf_var_name =  "$yaml_var" >> $filename
    fi
}

add_tfvars "root_volume_size" "$(eval echo '$'cluster_cloud_provisioner_node_group_${i}_root_volume_size)"
```

Reply:
I was thinking of this first, but there is conditions which resolved inside - some variables should processed with terminated \", some of variables should be processed using other functions like   to_tf_list 
```
eval override_instance_types='$'cluster_cloud_provisioner_node_group_${i}_override_instance_types; if [ ! -z "$override_instance_types" ]; then echo override_instance_types = $(to_tf_list "$override_instance_types")  >> worker_groups.tfvars; fi
                    eval spot_allocation_strategy='$'cluster_cloud_provisioner_node_group_${i}_spot_allocation_strategy; if [ ! -z $spot_allocation_strategy ]; then echo spot_allocation_strategy =  \""$spot_allocation_strategy"\"  >> worker_groups.tfvars; fi
```
So I propose to keep it as is, because auxilary function would just increase complexity and tracebility

Reply:
Wrong indentation - fixed